schemas/ingest-metrics.v1.schema.json

```diff
@@ -0,0 +1,73 @@
+{
+  "$schema": "http://json-schema.org/draft-07/schema#",
+  "$ref": "#/definitions/Main",
+  "definitions": {
+    "Main": {
+      "type": "object",
+      "title": "ingest_metric",
+      "additionalProperties": false,
+      "properties": {
+        "org_id": {
+          "description": "The organization for which this metric is being sent.",
+          "type": "integer"
+        },
+        "project_id": {
+          "description": "The project for which this metric is being sent.",
+          "type": "integer"
+        },
+        "name": {
+          "description": "The metric name. Relay sometimes calls this an MRI and makes assumptions about its string shape, and those assumptions also exist in certain queries. The rest of the ingestion pipeline treats it as an opaque string.",
+          "type": "string"
+        },
+        "type": {
+          "description": "The metric type. [c]ounter, [d]istribution, [s]et. Relay additionally defines Gauge, but that metric type is completely unsupported downstream.",
+          "type": "string",
+          "pattern": "^[cds]$"
+        },
+        "timestamp": {
+          "description": "The timestamp at which this metric was being sent. Relay will round this down to the next 10-second interval.",
+          "type": "integer"
+        },
+        "tags": {
+          "$ref": "#/definitions/StringToString"
+        },
+        "value": {
+          "anyOf": [
+            {
+              "type": "integer"
+            },
+            {
+              "type": "array",
+              "items": {
+                "type": "number"
+              }
```

Comment:
Is there a way to tell json schema that we have either `"type": "c"` with `"value": <int>` or `"type": "^[ds]$"` with `"value": <array>`? Maybe by having two different subschemas?

Reply:
I believe that might be possible using `allOf`, but I am not sure we want to use that construct (because it produces worse types). Also technically the metrics indexer does not care about it. We could consider it for the snuba topics